sdk/samples/ping_app.c

```diff
@@ -0,0 +1,400 @@
+            break;
+        case 'f':
+            config.vid_fps = atof(optarg);
+            break;
+        case 'r':
+            strlcpy(config.recv_addr, optarg, sizeof(config.recv_addr));
+            break;
+        case 'i':
+            strlcpy(config.recv_port, optarg, sizeof(config.recv_port));
+            break;
+        case 's':
+            strlcpy(config.send_addr, optarg, sizeof(config.send_addr));
+            break;
+        case 'p':
+            strlcpy(config.send_port, optarg, sizeof(config.send_port));
+            break;
+        case 'o':
+            strlcpy(config.protocol_type, optarg, sizeof(config.protocol_type));
+            break;
+        case 'n':
+            config.total_num = atoi(optarg);
+            break;
+        case 't':
+            strlcpy(config.payload_type, optarg, sizeof(config.payload_type));
+            break;
+        case 'k':
+            strlcpy(config.socket_path, optarg, sizeof(config.socket_path));
+            break;
+        case 'd':
+            config.interface_id = atoi(optarg);
+            break;
+        case 'l':
+            config.loop = (atoi(optarg)>0);
+            break;
+        case 'x':
+            strlcpy(config.pix_fmt_string, optarg, sizeof(config.pix_fmt_string));
+            break;
+        case 'm':
+            config.threads_num = atoi(optarg);
+            break;
+        case '?':
+            usage(stderr, argv[0]);
+            return 1;
+        default:
+            break;
+        }
+    }
+
+    if (help_flag) {
+        usage(stdout, argv[0]);
+        return 0;
+    }
+    signal(SIGINT, intHandler);
+
+    err = mesh_create_client(&client, NULL);
+    if (err) {
+        printf("Failed to create a mesh client: %s (%d)\n",
+               mesh_err2str(err), err);
+        exit(-1);
+    }
+
+    pthread_t *sender_threads = malloc(config.threads_num * sizeof(pthread_t));
+    pthread_t *receiver_threads = malloc(config.threads_num * sizeof(pthread_t));
+    t_data *recv_threads_data = malloc(config.threads_num * sizeof(t_data));
+    int *sender_threads_id = malloc(config.threads_num * sizeof(int));
+
+    for (int i = 0; i < config.threads_num; i++) {
+        recv_threads_data[i].latency_results = malloc(sizeof(*(recv_threads_data[i].latency_results)) * PRINT_COUNT);
+        recv_threads_data[i].thread_id = i;
+        if (pthread_create(&receiver_threads[i], NULL, receiver_thread, &recv_threads_data[i]) != 0) {
+            perror("pthread_create");
+            return 1;
+        }
+        usleep(100000);
+        sender_threads_id[i] = i;
+        if (pthread_create(&sender_threads[i], NULL, sender_thread, &sender_threads_id[i]) != 0) {
+            perror("pthread_create");
+            return 1;
+        }
+        usleep(100000);
+    }
+
+    // time to setup
+    sleep(1);
+
+    // Increment atomic counter and signal threads
+    for (int i = 1; i <= PRINT_COUNT; i++) {
+        usleep(500000);  // Delay to synchronize output
```

Comment:
Delay should be based on vid_fps

Reply:
ok, it can be, but here this parameter will serve different purpose. It those tests I just wanted to test latency without using full bandwidth, I just put big magic number as a delay between buffers, but I can parametrize this and allow the user to specify this

Reply:
done